components/script/dom/htmlelement.rs

```diff
@@ -839,10 +904,97 @@ impl VirtualMethods for HTMLElement {
                     DOMString::from(&**attr.value()),
                 );
             },
+            (&local_name!("form"), mutation) => {
+                if self.is_form_associated_custom_element() {
+                    self.form_attribute_mutated(mutation);
+                }
+            },
+            (&local_name!("disabled"), AttributeMutation::Set(_)) => {
+                // Adding a "disabled" attribute disables an enabled form element.
+                if self.is_form_associated_custom_element() {
```

Comment:
I would move the the definition of `element` outside the match and do this:

```suggestion
             // Adding a "disabled" attribute disables an enabled form element.
            (&local_name!("disabled"), AttributeMutation::Set(_)) if self.is_form_associated_custom_element() && element.enabled_state() => {
```